src/module/item/consumable/document.ts

```diff
@@ -74,18 +75,33 @@ class ConsumablePF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extend
 
         this.system.uses.max ||= 1;
 
-        // Refuse to serve rule elements if this item is ammunition and has types that perform writes
-        if (!this.isAmmo) return;
-        for (const rule of this.system.rules) {
-            if (rule.key === "RollOption" && "toggleable" in rule && !!rule.toggleable) {
-                console.warn("Toggleable RollOption rule elements may not be added to ammunition");
-                this.system.rules = [];
-                break;
-            } else if (["GrantItem", "ChoiceSet"].includes(String(rule.key))) {
-                console.warn(`${rule.key} rule elements may not be added to ammunition`);
-                this.system.rules = [];
-                break;
+        if (this.isAmmo) {
```

Comment:
Return early if it isn't ammo to spare the nesting level

Reply:
My concern with doing that is that doing an early return of that sort on a more generic function for a specific use case can cause a potential issue where we don't notice the early return when we add more to it. Its a mistake we've done before, although in the sheets.

Perhaps a private function for prepare ammo data?

Reply:
Secondary method is fine

Reply:
Done, though untested. If I remember, using actual private with our proxy stuff didn't work.